serverless/serverless-permission-boundary.yaml

```diff
@@ -18,35 +22,63 @@ Resources:
               - iam:SimulatePrincipalPolicy
             Resource:
               - "*"
+          # These are for serverless services that need a * in resources as we do not know
+          # the exact path of in advance e.g. api gateway. 
           - Sid: AllowServerlessServices
             Effect: Allow
             Action:
               - apigateway:*
               - cloudwatch:*
-              - cognito-idp:*
```

Comment:
I like how you've decreased the scope of some of the services by including the Squad name, but I'm not sure it's always a good idea to limit the service. The permissions in a permission boundary only take effect if the underlying CI user's permission _also_ has the permission granted. I'll just have a check on these.